### PR TITLE
C pathfinding and chokes

### PR DESCRIPTION
**Pull request by spudde123**
- Implementing pathfinding and map analysis in C, allowing to remove pyastar and sc2pathlib as dependencies
- Implementing smoothed pathfinding
- Fixing choke objects in the mapanalyzer so they have sides etc that better reflect reality
- Fixing ramp objects that come from burnysc2 if the info is broken due to destructables

### CODE REVIEW

File: MapAnalyzer/utils.py [hunk 1]
```diff
@@ -27,7 +63,7 @@ def get_sets_with_mutual_elements(list_mdchokes: List[PathLibChoke],
     for c in list_mdchokes:
         s2 = c.pixels
         s3 = s1 ^ s2
-        if len(s3) != (len(s1) + len(s2)):
+        if len(s3) <= 0.95*(len(s1) + len(s2)):
```
Comment (eladyaniv01) [inline MapAnalyzer/utils.py:66]: I remember we discussed this change,  but i forgot why :)
 
please comment explaining why a factor of 0.95 was added here

File: pf_perf.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+map_data.plot_influenced_path(start=p0, goal=p1, weight_array=arr, allow_diagonal=True)
+
+start = time.clock()
+path2 = map_data.pathfind_c(p0, p1, grid=arr)
+ext_time = time.clock() - start
+print("extension astar time: {}".format(ext_time))
+print("div: {}".format(ext_time / pyastar_time))
+
+map_data.plot_influenced_path_c(start=p0, goal=p1, weight_array=arr)
```
Comment (eladyaniv01) [inline pf_perf.py:71]: this can be used for benchmarks , right ?

**Review by eladyaniv01 — COMMENTED** [review]
great work,   i posted some comments,  mostly regarding naming conventions  and how the lib is going to treat the c ext,   as  its main feature, and not some added feature (in my opinion  features from Cext shouldn't be names specially,  only features from external sources should)

File: MapAnalyzer/utils.py (hunk 1 above)
Comment (spudde123) [reply MapAnalyzer/utils.py:66]: The code is run to check if the detected chokes overlap with ramps or vision blockers. Previously if they had any overlap at all they got thrown out, but there can be legitimate chokes that start from near a ramp for example and have a few pixels in common so I added a small threshold. Probably could come up with some better criteria but it seems to work okay in practice

**Pull request merged**